### PR TITLE
Tidying up

### PR DESCRIPTION
**Pull request by bell-pa**
## Changes in this PR
* Update README.md files
* Remove unused scripts
* Fix Gradle bootRunLocal task
* Update Gradle. v8.3 to 8.4
* Update gradle-spring-boot plugin. 5.4.1 to 5.5.0
* Update Json WebToken dependency. 0.11.5 to 0.12.2
* Fix a Gradle deprecated feature warning: Switch to using Test Suites.

No deployable code changes in this  PR.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,55 +5,68 @@
...
 ## Running the tests
 
-To run linting and tests, run:
+To run linting and tests, do:
 
 ```bash
-script/test
+./gradlew clean build
 ```
+Repository integration tests ruse an embedded H2 database. REST API tests start a local server which listens on a random
```
Comment (nsmj63-dxw) [inline README.md:63]: Typo here - `use` over `ruse` :)

File: script/local-scripts/README.md [hunk 2]
```diff
@@ -18,16 +18,16 @@ running in docker and prints it to stdout.
...
-download the CSV data to files,  edit in a spreadsheet program and ensure the changes are save
+download the CSV data to files,  edit in a spreadsheet program (or IntelliJ's built-in CSV editor) and ensure the changes are saved
 in CSV format. Upload the modified file(s) using the corresponding `put-courses`, `put-offerings` or `put-prereq`
-scripts.  Courses variants and Offerings are soft-deleted when absent from the uploaded CSV.
+scripts.  Course variants and Offerings are soft-deleted when absent from the uploaded CSV.
 Returning a Course variant or Offering to the CSV will restore the soft-deleted course/offering in the application.
 
-Courses are distinguished by the value in the `identifier` column. All other values may be changed
-and will be applied to the course having that identifier.
+Courses are distinguished by the value in the `identifier` column. Any other other column value may be changed
```
Comment (nsmj63-dxw) [inline script/local-scripts/README.md:26]: You've repeated `other` here :)

File: script/local-scripts/README.md [hunk 3]
```diff
@@ -18,16 +18,16 @@ running in docker and prints it to stdout.
...
+and the change will be applied to the course with that identifier.
 
-Offerings are distinguished by course identifier and prisonId.  All other values may be changed and will
+Offerings are distinguished by course identifier and prisonId.  any other value may be changed and the change will
```
Comment (nsmj63-dxw) [inline script/local-scripts/README.md:29]: Capital `A` on `any`.

**Review by nsmj63-dxw — CHANGES_REQUESTED** [review]

File: script/local-scripts/README.md (hunk 2 above)
Comment (bell-pa) [reply script/local-scripts/README.md]: Fixed

File: README.md (hunk 1 above)
Comment (bell-pa) [reply README.md]: Fixed

**Review by nsmj63-dxw — APPROVED** [review]

**Pull request merged**